packages/core/src/transport/transport.ts

```diff
@@ -29,9 +64,32 @@ export class HttpRequest {
         reportBeaconError(e)
       }
     }
-    const request = new XMLHttpRequest()
-    request.open('POST', url, true)
-    request.send(data)
+
+    const transportErrorHandler = (ctx: any) => {
+      const transportErrCtx: TransportErrorCtx = {
+        on_line: navigator.onLine,
+        beacon: {
+          size,
+          url,
+          try_beacon: tryBeacon,
+          bytes_limit: this.bytesLimit,
+        },
+        ...ctx,
+      }
+      addMonitoringMessage('Internal XHR failed', transportErrCtx)
+    }
+
+    try {
+      const request = new XMLHttpRequest()
+      request.addEventListener('error', (e) => transportErrorHandler(getTransportErrorEventDetails(e)))
```

Comment:
Beside listening to the “error” event, you should also listening to “load” and send a message if the status is >= 400 or the response is not `{}` , something like that. It could be useful to include `request.responseText` in the context too

Reply:
Good idea. Let's see how this works out first?

Reply:
As needed, we can add to the XHR (and something similar to sendBeacon)
- onTimeout
- onLoad
- onAbort

Reply:
According to the Jira task, we want to make it easier to investigate issues where the intake rejects the request we are sending because it is too large. So, your PR does not fit our current needs: when the intake rejects the request, the XHR succeeds but the response will have an invalid status code and/or an error message in the body I guess. So we’ll have to hook into the “onload” event.

`ontimeout`, `onabort`, `onerror` and `try/catch` are nice to have but a bit out of scope for this PR. I suggest to keep what you did (`onerror` and `try/catch`), add a check on the response via `onload`. That should be enough for now. Does it make sense?